pkg/oc/cli/cancelbuild/cancelbuild.go

```diff
@@ -244,9 +244,18 @@ func (o *CancelBuildOptions) RunCancelBuild() error {
 				o.ReportError(fmt.Errorf("build %s/%s failed to update: %v", build.Namespace, build.Name, err))
 				return
 			}
+			statusMsg := &printers.NamePrinter{Operation: "marked for cancellation, waiting to be cancelled"}
```

Comment:
You can't create printer like that, can you use the `o.PrinterCancel.PrintObj` here?

Reply:
I would have to manipulate the Operation field between calls to `PrintObj` to  change the message from `marked for cancellation, waiting to be cancelled` to
`cancelled`

Given the FIXME comment above complaining about the multiple printers for each message, I tried to not create yet another Printer.

But if modifying the Operation field is stylistically preferred I'll make the change.

Also note:  this method also leverages `fmt.Fprintf(...` in some cases but I consciously did not go down that path in order to get the `build.build.openshift.io/sample-pipeline-1...` naming to be consistent with the typical, everything is working, path.

Reply:
> Also note: this method also leverages fmt.Fprintf(... in some cases but I consciously did not go down that path in order to get the build.build.openshift.io/sample-pipeline-1... naming to be consistent with the typical, everything is working, path.

+1

> Given the FIXME comment above complaining about the multiple printers for each message, I tried to not create yet another Printer.

Yeah, this is something we hope to improve in the near future. For now, it's fine to keep creating new printers if a different message is needed

Reply:
pushed printer refactor